packages/fuels-abigen-macro/tests/harness.rs

```diff
@@ -1629,3 +1629,32 @@ async fn test_transaction_script_workflow() {
     let response = call_handler.get_response(receipts).unwrap();
     assert_eq!(response.value, 42);
 }
+
+#[tokio::test]
+async fn nested_enums() {
```

Comment:
this test doesn't test for using nested enums as input, but only as output. To check for encoding, please take inspiration from the `enum_input` tests in PR #361 (it's actually sometimes not enough to just get them in then out).

Reply:
Thanks for the great review. I have updated both the unit and the harness tests. We check now for input and output of nested enums with depth 3, as was done in https://github.com/FuelLabs/fuels-rs/pull/361.